solutions/USACO open16-bull-in-china-shop.mdx

```diff
@@ -0,0 +1,135 @@
+---
+id: usaco-640
+source: USACO Bronze 2016 Open
+title: Bull in a China Shop
+author: Maggie Liu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_bcs_bronze_open16.html)
+
+## Explanation
+
+We can loop through all possible pairs of pieces and shift them to check if they can make the original figurine. Since we cannot shift a piece so any '#' fall outside the $N \times N$ grid, we need to find the sides of the pieces to determine how far to shift them.
+
+<Warning>
+
+The official solution doesn't check if any '#' fall outside the $N \times N$ grid, which is wrong.
```

Comment:
You are right the original solution seems wrong. Although it is possible that we are missing something about the statement.